apiserver/pkg/interceptor/interceptor.go

```diff
@@ -19,3 +21,40 @@ func APIServerInterceptor(ctx context.Context, req interface{}, info *grpc.Unary
 	klog.Infof("%v handler finished", info.FullMethod)
 	return
 }
+
+func TimeoutInterceptor(timeout time.Duration) grpc.UnaryServerInterceptor {
```

Comment:
Is There an integration test we can add to make sure this is working as expected ?